src/rx5808-pro-diversity/state_settings_rssi.cpp

```diff
@@ -0,0 +1,178 @@
+        break;
+
+        case InternalState::SCANNING_HIGH:
+            if (Receiver::rssiARaw > EepromSettings.rssiAMax)
+                EepromSettings.rssiAMax = Receiver::rssiARaw;
+
+            #ifdef USE_DIVERSITY
+                if (Receiver::rssiBRaw > EepromSettings.rssiBMax)
+                    EepromSettings.rssiBMax = Receiver::rssiBRaw;
+            #endif
+        break;
+    }
+
+    Receiver::setChannel((Receiver::activeChannel + 1) % CHANNELS_SIZE);
+    if (Receiver::activeChannel == 0) {
+        currentSweep++;
+
+        if (currentSweep == RSSI_SETUP_RUN) {
+            switch (internalState) {
+                case InternalState::SCANNING_LOW:
+                    internalState = InternalState::WAIT_FOR_HIGH;
+                break;
+
+                case InternalState::SCANNING_HIGH:
+                    internalState = InternalState::DONE;
+                break;
+            }
+
+            Ui::needUpdate();
+        }
+    }
+}
+
+void StateMachine::SettingsRssiStateHandler::onButtonChange(
+    Button button,
+    Buttons::PressType pressType
+) {
+    if (button != Button::MODE || pressType != Buttons::PressType::SHORT)
+        return;
+
+    switch (internalState) {
+        case InternalState::WAIT_FOR_LOW:
+            internalState = InternalState::SCANNING_LOW;
+            currentSweep = 0;
+            Receiver::setChannel(0);
+
+            EepromSettings.rssiAMin = UINT16_MAX;
+            #ifdef USE_DIVERSITY
+                EepromSettings.rssiBMin = UINT16_MAX;
+            #endif
+        break;
+
+        case InternalState::WAIT_FOR_HIGH:
+            internalState = InternalState::SCANNING_HIGH;
+            currentSweep = 0;
+            Receiver::setChannel(0);
+
+            EepromSettings.rssiAMax = 0;
+            #ifdef USE_DIVERSITY
+                EepromSettings.rssiBMax = 0;
+            #endif
+        break;
+
+        case InternalState::DONE:
+            EepromSettings.save();
+            StateMachine::switchState(StateMachine::State::MENU);
+        break;
+    }
+
+    Ui::needUpdate();
+}
+
+
+void StateMachine::SettingsRssiStateHandler::onInitialDraw() {
+    Ui::needUpdate(); // Lazy. :(
+}
+
+void StateMachine::SettingsRssiStateHandler::onUpdateDraw() {
+    Ui::clear();
+
+    switch (internalState) {
+        case InternalState::WAIT_FOR_LOW:
+            Ui::display.setTextSize(1);
+            Ui::display.setCursor(0, 0);
+            Ui::display.print(PSTR2("1/4\nTurn off all VTXs."));
+            Ui::display.setCursor(0, (CHAR_HEIGHT + 1) * 2);
+            Ui::display.print(PSTR2("Remove RX antennas."));
+
+            Ui::display.setCursor(0, SCREEN_HEIGHT - CHAR_HEIGHT - 1);
+            Ui::display.print(PSTR2("Press MODE when ready."));
+        break;
+
+        case InternalState::SCANNING_LOW:
+            Ui::display.setTextSize(1);
+            Ui::display.setCursor(0, 0);
+            Ui::display.print(PSTR2("2/4\nScanning for lowest\nRSSI..."));
```

Comment:
Yeah, for sure! The entire calibration screen is garbage at the moment, hoping to spruce it up with less text. :smile: